lang/cs/Org.Apache.REEF.Tests/Functional/FaultTolerant/TestContextStart.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using Org.Apache.REEF.Common.Context;
+using Org.Apache.REEF.Common.Events;
+using Org.Apache.REEF.Common.Tasks;
+using Org.Apache.REEF.Driver;
+using Org.Apache.REEF.Driver.Context;
+using Org.Apache.REEF.Driver.Evaluator;
+using Org.Apache.REEF.Tang.Annotations;
+using Org.Apache.REEF.Tang.Interface;
+using Org.Apache.REEF.Tang.Util;
+using Org.Apache.REEF.Utilities.Logging;
+using Xunit;
+
+namespace Org.Apache.REEF.Tests.Functional.FaultTolerant
+{
+    /// <summary>
+    /// This test case servers as an example to put data downloading at part of the ContextStartHandler
+    /// </summary>
+    public class TestContextStart : ReefFunctionalTest
+    {
+        private static readonly Logger Logger = Logger.GetLogger(typeof(TestContextStart));
+        private const string StartedHandlerMessage = "Start Handler is called.";
+        private const string DataDownLoadStartedMessage = "Data Downloading started.";
+        private const string DataDownLoadCompletedMessage = "Data download completed.";
+
+        public TestContextStart()
+        {
+            Init();
```

Comment:
The convention is to put test initialization in class constructor.
I have some ideas about cleaning up tests inherited from ReefFunctionalTest: all constructors call Init() so they can be replaced with base constructor calling Init, and same about Dispose(). But I want to postpone this change until our AppVeyor config is good, so that we have more validation of this approach.